test/apps/iodemo/io_demo.cc

```diff
@@ -965,6 +982,11 @@ class DemoClient : public P2pDemoCommon {
             LOG << "Connected to " << server_name(server_index);
         }
 
+        /* all servers are connected or in progress, reset status to OK */
+        if (_active_servers.size() == _server_info.size()) {
+            _status = OK;
+        }
```

Comment:
can we avoid this? set status to !=OK only when the detected error is really fatal